### ISSUE TITLE
Strange cache problem with composer's updater

### ISSUE THREAD

**Issue by Taluu** [composer/composer#4378]
Due to a mistake I made (pushed something a wrong repository), the packages on packagist was wrong (basically, I pushed the wrong package on the wrong repository for 1 minute or 2).

So packagist had updated its information displaying the information from another package, but I fixed that soon after I saw what I did.

On machines that did not update the package, it is fine, but on my machine, composer seems to persistly have a wrong autoloader, not loading the files as it should...

In my `autoload_psr4.php`, I have the following :

```php
<?php

// autoload_psr4.php @generated by Composer

$vendorDir = dirname(dirname(__FILE__));
$baseDir = dirname($vendorDir);

return array(
  //...
    'CalendArt\\Adapter\\Office365\\' => array($vendorDir . '/calendart/office365-adapter/src', $vendorDir . '/calendart/office365-adapter/test'),
    'CalendArt\\' => array($vendorDir . '/calendart/calendart/src', $vendorDir . '/calendart/calendart/test', $vendorDir . '/calendart/google-adapter/src', $vendorDir . '/calendart/google-adapter/test'),
);
```

When I should have :

```php
<?php

// autoload_psr4.php @generated by Composer

$vendorDir = dirname(dirname(__FILE__));
$baseDir = dirname($vendorDir);

return array(
  //...
    'CalendArt\\Adapter\\Office365\\' => array($vendorDir . '/calendart/office365-adapter/src', $vendorDir . '/calendart/office365-adapter/test'),
    'CalendArt\\Adapter\\Office365\\' => array($vendorDir . '/calendart/office365-adapter/src', $vendorDir . '/calendart/office365-adapter/test'),
    'CalendArt\\' => array($vendorDir . '/calendart/calendart/src', $vendorDir . '/calendart/calendart/test'),
);
```

And in my composer.lock, **each time** I am making an update, I have also the wrong package information (the ones from calendart/calendart instead of the ones from calendart/google-adapter). I guess there is some sort of cache, but this persists even though I deleted composer cache from `~/.composer`...

But, as I already said, this problem doesn't seem to occur on another machine, and the package calendart/google-adapter on packagist looks fine too.

Any idea what could be provoking this ?

Thanks.

**Comment by alcohol** [comment]
Delete your lock file and clear your cache.

**Comment by Taluu** [comment]
I can't delete the lock file, as it is the lock file for our project (we are an application, thus we need a fixed lock file). I already tried when clearing the cache, the autoload is still not correct.

And whenever I'm doing a new update on that package, the package information from the other package is still there instead of what it should be. And this is happening only on my machine, even when clearing the cache in `~/.composer/cache`...

**Comment by alcohol** [comment]
Sure you can delete the lock file. `rm composer.lock && rm -rf vendor/`. Then just run install to generate a new one.

**Comment by Taluu** [comment]
Once again, I can't, as it is committed on our repository so it basically needs to stay (exactly) the same, with the same versions. And besides, this same composer.lock file (once I fix it manually...) is working normally (even after an update) on other machines, so I'm pretty sure it is a cache somewhere else than on my `.composer` directory

**Comment by Taluu** [comment]
But removing the vendor directory and the cache seems to do the trick. Closing.

**Comment by alcohol** [comment]
How can you run update? That modifies the .lock file too. Your statements don't really make sense :-(

**Comment by Taluu** [comment]
We are updating package by package, and once I modified my package, I needed to reflect the changes on the main project. But I messed up with git, and once I fix'd that, composer went haywire (only on my machine though), and persisted into changing the information in the lock file into a wrong package information. Even though on packagist, after 2 minutes (or so), the information was back to what it was supposed to be.

So it basically generated a wrong autoloader.
